### PR TITLE
[babel__traverse] Add new method `hasNode()` to `NodePath`

### PR DESCRIPTION
**Pull request by danez**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/babel/babel/pull/13940
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Some prettier style changes also sneaked in 😊

### CODE REVIEW

File: types/babel__traverse/index.d.ts [hunk 1]
```diff
@@ -140,7 +140,12 @@ export class Scope {
 
     crawl(): void;
 
-    push(opts: { id: t.LVal; init?: t.Expression | undefined; unique?: boolean | undefined; kind?: 'var' | 'let' | 'const' | undefined }): void;
+    push(opts: {
+        id: t.LVal;
+        init?: t.Expression | undefined;
+        unique?: boolean | undefined;
+        kind?: 'var' | 'let' | 'const' | undefined;
+    }): void;
```
Comment (danez) [inline types/babel__traverse/index.d.ts:148]: prettier

File: types/babel__traverse/index.d.ts [hunk 2]
```diff
@@ -194,13 +199,11 @@ export class Binding {
...
-        [Type in Node['type']]?: VisitNode<S, Extract<Node, { type: Type }>>;
-    } &
-    {
-        [K in keyof t.Aliases]?: VisitNode<S, t.Aliases[K]>;
-    };
+export type Visitor<S = {}> = VisitNodeObject<S, Node> & {
+    [Type in Node['type']]?: VisitNode<S, Extract<Node, { type: Type }>>;
+} & {
+    [K in keyof t.Aliases]?: VisitNode<S, t.Aliases[K]>;
+};
```
Comment (danez) [inline types/babel__traverse/index.d.ts:206]: prettier

File: types/babel__traverse/ts4.1/index.d.ts [hunk 3]
```diff
@@ -182,13 +182,11 @@ export class Binding {
...
-        [Type in Node['type']]?: VisitNode<S, Extract<Node, { type: Type }>>;
-    } &
-    {
-        [K in keyof t.Aliases]?: VisitNode<S, t.Aliases[K]>;
-    };
+export type Visitor<S = {}> = VisitNodeObject<S, Node> & {
+    [Type in Node['type']]?: VisitNode<S, Extract<Node, { type: Type }>>;
+} & {
+    [K in keyof t.Aliases]?: VisitNode<S, t.Aliases[K]>;
+};
```
Comment (danez) [inline types/babel__traverse/ts4.1/index.d.ts:189]: prettier

**Review by amcasey — COMMENTED** [review]
LGTM, but I'd like to hear from an owner.

**Review by jakebailey — APPROVED** [review]
Type annotation matches that in the linked PR (though I do wonder why the project doesn't ship types if upstream is TS already, oh well).

**Comment by danez** [comment]
Ready to merge

**Pull request merged**